var/spack/repos/builtin/packages/blast-plus/package.py

```diff
@@ -79,6 +80,7 @@ def patch(self):
     depends_on("bzip2", when="+bzip2")
     depends_on("lzo", when="+lzo")
     depends_on("pcre", when="+pcre")
+    depends_on("llvm-openmp", when="%apple-clang")
```

Comment:
We have https://github.com/spack/spack/pull/42969 which would solve the issue more generally. It's blocked out of the concerns raised by the fact that `openmp` is explicitly disabled by Apple Clang. 

@trws Do you have opinions on this?